messenger/http_transporter.go

```diff
@@ -124,13 +139,12 @@ type HTTPTransporter struct {
 
 // NewHTTPTransporter creates a new http transporter with an optional binding address.
 func NewHTTPTransporter(upid upid.UPID, address net.IP) *HTTPTransporter {
-	tr := &http.Transport{}
 	result := &HTTPTransporter{
 		upid:         upid,
 		messageQueue: make(chan *Message, defaultQueueSize),
 		mux:          http.NewServeMux(),
-		client:       &http.Client{Transport: tr},
-		tr:           tr,
+		client:       httpClient,
+		tr:           httpTransport,
```

Comment:
Why is the transport supplied to the HTTPTransporter if you can just pull it out of the Client with `t.client.Transport`?